opensearch-operator/pkg/reconcilers/securityconfig.go

```diff
@@ -0,0 +1,208 @@
+package reconcilers
+
+import (
+	"context"
+	"crypto/sha1"
+	"encoding/base64"
+	"fmt"
+	"sort"
+	"time"
+
+	"github.com/banzaicloud/operator-tools/pkg/reconciler"
+	"github.com/go-logr/logr"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/record"
+	opsterv1 "opensearch.opster.io/api/v1"
+	"opensearch.opster.io/pkg/builders"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const (
+	checksumAnnotation = "securityconfig/checksum"
+)
+
+type SecurityconfigReconciler struct {
+	reconciler.ResourceReconciler
+	client.Client
+	ctx               context.Context
+	recorder          record.EventRecorder
+	reconcilerContext *ReconcilerContext
+	instance          *opsterv1.OpenSearchCluster
+	logger            logr.Logger
+}
+
+func NewSecurityconfigReconciler(
+	client client.Client,
+	ctx context.Context,
+	recorder record.EventRecorder,
+	reconcilerContext *ReconcilerContext,
+	instance *opsterv1.OpenSearchCluster,
+	opts ...reconciler.ResourceReconcilerOption,
+) *SecurityconfigReconciler {
+	return &SecurityconfigReconciler{
+		Client: client,
+		ResourceReconciler: reconciler.NewReconcilerWith(client,
+			append(opts, reconciler.WithLog(log.FromContext(ctx).WithValues("reconciler", "securityconfig")))...),
+		ctx:               ctx,
+		reconcilerContext: reconcilerContext,
+		recorder:          recorder,
+		instance:          instance,
+		logger:            log.FromContext(ctx),
+	}
+}
+
+func (r *SecurityconfigReconciler) Reconcile() (ctrl.Result, error) {
+	if r.instance.Spec.Security == nil || r.instance.Spec.Security.Config == nil || r.instance.Spec.Security.Config.SecurityconfigSecret.Name == "" {
+		return ctrl.Result{}, nil
+	}
+	namespace := r.instance.Namespace
+	clusterName := r.instance.Name
+	jobName := clusterName + "-securityconfig-update"
+	configSecretName := r.instance.Spec.Security.Config.SecurityconfigSecret.Name
+	adminCertName := r.determineAdminSecret()
+
+	if adminCertName == "" {
+		r.logger.Info("Cluster is running with demo certificates. Skipping securityconfig")
+		return ctrl.Result{}, nil
+	}
+
+	// Wait for secret to be available
+	configSecret := corev1.Secret{}
+	if err := r.Get(r.ctx, client.ObjectKey{Name: configSecretName, Namespace: namespace}, &configSecret); err != nil {
```

Comment:
May be cleaner to put a little more logic here.  E.g. if the error is `ErrIsNotFound` then we can requeue without emitting the error (maybe add an info or debug log), but if the err is any other type treat it as an unexpected error

Reply:
Good idea, changed it.